test/sequential/test-heapdump.js

```diff
@@ -25,13 +25,15 @@ process.chdir(tmpdir.path);
 }
 
 {
-  const readonlyFile = 'ro';
-  fs.writeFileSync(readonlyFile, Buffer.alloc(0), { mode: 0o444 });
+  const directory = 'directory';
+  fs.mkdirSync(directory);
```

Comment:
I don't think this change will keep the test content as same as before

Reply:
@mawaregetsuka Yes but the intention is still the same. If you go through the PR description where this was introduced - https://github.com/nodejs/node/pull/41373, it mentions:
> this PR makes v8.writeHeapSnapshot throw if the file could not be written

So the current test still attempts to recreate a scenario where the function won't be able to write to the file, so we expect it to throw.

Reply:
Thank you, now I understand. That's a clever solution